File: modern/_1.20.1/src/main/java/cn/enaium/noexpensive/command/MaxLevelCommand.java
package cn.enaium.noexpensive.command;

import cn.enaium.noexpensive.Config;
import com.mojang.brigadier.Command;
import com.mojang.brigadier.CommandDispatcher;
import com.mojang.brigadier.arguments.IntegerArgumentType;
import net.minecraft.server.command.CommandManager;
import net.minecraft.server.command.ServerCommandSource;
import net.minecraft.text.Text;

import static cn.enaium.noexpensive.NoExpensive.ROOT;

/**
 * @author Enaium
 */
public class MaxLevelCommand {
    public static void register(CommandDispatcher<ServerCommandSource> dispatcher) {
        dispatcher.register(ROOT.then(CommandManager.literal("maxLevel").executes(context -> {
            context.getSource().sendFeedback(() -> Text.translatable("command.maxLevel.get", Config.getModel().maxLevel), false);
            return Command.SINGLE_SUCCESS;
        }).then(CommandManager.argument("level", IntegerArgumentType.integer()).executes(context -> {
            Config.getModel().maxLevel = IntegerArgumentType.getInteger(context, "level");
            context.getSource().sendFeedback(() -> Text.translatable("command.maxLevel.success", IntegerArgumentType.getInteger(context, "level")), false);
            Config.save();
            return Command.SINGLE_SUCCESS;
        }))));
    }
}

File: modern/_1.19.4/src/main/kotlin/cn/enaium/noexpensive/command/MaxLevelCommand.kt
package cn.enaium.noexpensive.command

import cn.enaium.noexpensive.Config
import cn.enaium.noexpensive.ROOT
import com.mojang.brigadier.Command
import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.arguments.IntegerArgumentType
import com.mojang.brigadier.context.CommandContext
import net.minecraft.server.command.CommandManager
import net.minecraft.server.command.ServerCommandSource
import net.minecraft.text.Text

/**
 * @author Enaium
 */
fun maxLevelCommand(dispatcher: CommandDispatcher<ServerCommandSource>) {
    dispatcher.register(
        ROOT.then(
            CommandManager.literal("maxLevel").executes { context: CommandContext<ServerCommandSource> ->
                context.source.sendFeedback(
                    Text.translatable("command.maxLevel.get", Config.model.maxLevel), false
                )
                Command.SINGLE_SUCCESS
            }.then(
                CommandManager.argument("level", IntegerArgumentType.integer())
                    .executes { context: CommandContext<ServerCommandSource> ->
                        Config.model.maxLevel = IntegerArgumentType.getInteger(context, "level")
                        context.source.sendFeedback(
                            Text.translatable(
                                "command.maxLevel.success",
                                IntegerArgumentType.getInteger(context, "level")
                            ), false
                        )
                        Config.save()
                        Command.SINGLE_SUCCESS
                    })
        )
    )
}